content/blog/how-to-deploy-application-across-hybrid-clouds-beyond-hpe-greenlake-for-private-cloud-enterprise.md

```diff
@@ -0,0 +1,269 @@
+frontend-external       LoadBalancer   172.20.16.223    a52d7c861c01c4466803a44373bc11dc-1387384363.us-east-2.elb.amazonaws.com   80:31482/TCP                      40s   <none>                                    
+```
+
+### Deploy Application Microservices to PCE Private Cluster
+
+S﻿imilarly, create the namespace *boutique* in the cluster and then deploy 3 *Deployment* resources to the namespace:
+
+```markdown
+$ kubectl create ns boutique
+$ kubectl config set-context --current --namespace boutique
+$ kubectl apply -f k8s-manifests-deploy-private.yaml    
+deployment.apps/emailservice created
+deployment.apps/paymentservice created
+deployment.apps/shippingservice created
+```
+
+### Deploy Virtual Application Network
+
+Define the Virtual Application Network using *Skupper* on both AWS public and PCE private clusters:
+
+#### 1. In AWS public cluster, deploy the *aws-public* application router.
+
+```markdown
+$ kubectl config set-context --current –namespace boutique
+$ skupper init --site-name aws-public                                                                                         
+Waiting 115 seconds for LoadBalancer IP or hostname...                                         
+Waiting 111 seconds for LoadBalancer IP or hostname...                                         
+Waiting 108 seconds for LoadBalancer IP or hostname...                                        
+Skupper is now installed in namespace 'boutique'.  Use 'skupper status' to get more information.            
+                                                                        
+$ skupper status             
+Skupper is enabled for namespace "boutique" with site name "aws-public" in interior mode. It is connected to 1 other site. It has 10 exposed services.
+The site console url is:  https://aea867abf6fb6413d8f577652da564c1-130946084.us-east-2.elb.amazonaws.com:8080
+The credentials for internal console-auth mode are held in secret: 'skupper-console-users'
+```
+
+#### 2. In PCE private cluster, deploy the *pce-private* application router.
+
+```markdown
+$ kubectl config set-context --current –namespace boutique
+$ skupper init --ingress none --site-name pce-private
+Skupper is now installed in namespace 'boutique'.  Use 'skupper status' to get more information.
+
+$ skupper status
+Skupper is enabled for namespace "boutique" with site name "pce-private" in interior mode. It is not connected to any other sites. It has no exposed services
+```
+
+#### 3. In AWS public cluster, create a connection token for connection.
+
+```markdown
+$ skupper token create ~/aws-public-token.yaml                                                                              
+Token written to /home/guoping/aws-public-token.yaml                                           
+```
+
+#### 4. In PCE private cluster, define the connections to the AWS public cluster.
```

Comment:
In private Kubernetes cluster, define the connections to the public AWS EKS cluster.